### ISSUE TITLE
[QUESTION] Stream Screen through VideoCall apps

### ISSUE THREAD

**Issue by jonaspm** [webcamoid/webcamoid#50]
Can I use webcamoid to stream my desktop through videocall services like Skype or Google Hangouts? Because I already have tried using virtual webcam without luck.
Thanks in advance!

**Comment by hipersayanX** [comment]
Yes, virtual camera feature was made for that purpose.

>Because I already have tried using virtual webcam without luck.

Need more info. Which operating system? Skype and Google Hangouts didn't recognized the camera? Which other programs did you tested?. Please, include any other useful information.

**Comment by jonaspm** [comment]
Software: Webcamoid 7.2.1 x64 Windows installable
OS: Windows 10 x64 SL
Device: Laptop with HD Camera
I managed to make the virtual camera work with FaceFlow.com (see picture 1)
![faceflow](https://cloud.githubusercontent.com/assets/7599554/21033571/98fe2d04-bd6e-11e6-84ed-5f9a41daebae.png)

However, Hangouts does not detect the virtual camera for some reason. (see picture 2)
![webcam](https://cloud.githubusercontent.com/assets/7599554/21033522/1109fa40-bd6e-11e6-8c04-79492d459458.png)

**Comment by jonaspm** [comment]
Skype Preview for Windows 10 also doesn't detect ManyCam nor Webcamoid
![skype](https://cloud.githubusercontent.com/assets/7599554/21034190/f84a6b2a-bd72-11e6-822a-1b33bf02a3c2.PNG)

**Comment by hipersayanX** [comment]
I've implemented the virtual camera as a DirectShow filter (based on all the documentation out there), for that reason Firefox has not problem detecting it. However, my guess, Google Hangouts doesn't detect devices loaded as a DirectShow filter. 

The reason because Google Hangouts it's detecting the ManyCam camera is because they are probably implementing the virtual camera as a [KMDF o UMDF](https://en.wikipedia.org/wiki/Windows_Driver_Frameworks). I can try writing a UMDF driver, if MinGW allow me to do that, but KMDF requires code signing 💵 .

**Comment by hipersayanX** [comment]
Ok, I have good news!

I been experimenting with UMDF and KMDF drivers, and even when I was able to create a bare bone UMDF driver in GCC, at the end discarded the idea because not enough documentation to do something useful, and as I said before, KMDF requires signing, so these drivers are discarded.
Then I went for an [AVStream](https://github.com/Microsoft/Windows-driver-samples/tree/master/avstream/avshws) driver, the problem is that these drivers requires wdm.h, so these drivers are plain kernel drivers that also requires signing, so discarded too.

At end decided to install and test the virtual camera in Skype, and I can confirm that the **virtual webcam works out of the box in Skype**, but with one condition, **you must install the 32 bits version of Webcamoid**, and the reason is because, Skype as provided by Microsoft, is a 32 bits program only.

So I will update website and wiki, and close this. At the end it was a very easy to solve issue, I just over complicated things 😛
